### ISSUE TITLE
Fails to build on "buf.as_mut_ptr(), expected `u8`, found `i8`" on Linux on various archs

### ISSUE THREAD

**Issue by sylvestre** [byllyfish/exacl#107]
https://buildd.debian.org/status/package.php?p=rust-exacl&suite=sid
Fails on arm64, armel, armhf, ppc64el, s390x

```
error[E0308]: mismatched types
  --> src/unix.rs:42:17
   |
42 |                 buf.as_mut_ptr(),
   |                 ^^^^^^^^^^^^^^^^ expected `u8`, found `i8`
   |
   = note: expected raw pointer `*mut u8`
              found raw pointer `*mut i8`

error[E0308]: mismatched types
  --> src/unix.rs:86:17
   |
86 |                 buf.as_mut_ptr(),
   |                 ^^^^^^^^^^^^^^^^ expected `u8`, found `i8`
   |
   = note: expected raw pointer `*mut u8`
              found raw pointer `*mut i8`

error[E0308]: mismatched types
   --> src/unix.rs:129:17
    |
129 |                 buf.as_mut_ptr(),
    |                 ^^^^^^^^^^^^^^^^ expected `u8`, found `i8`
    |
    = note: expected raw pointer `*mut u8`
               found raw pointer `*mut i8`

error[E0308]: mismatched types
   --> src/unix.rs:167:17
    |
167 |                 buf.as_mut_ptr(),
    |                 ^^^^^^^^^^^^^^^^ expected `u8`, found `i8`
    |
    = note: expected raw pointer `*mut u8`
               found raw pointer `*mut i8`

For more information about this error, try `rustc --explain E0308`.
error: could not compile `exacl` due to 4 previous errors

```

**Comment by byllyfish** [comment]
Thanks for the report.

I will fix.

**Comment by sylvestre** [comment]
Super quick fix, bravo, I am impressed :)
Do you have an eta for a new version (or I can just apply your change on top of the current Debian package)

**Comment by byllyfish** [comment]
I released version 0.9.0 with the fix. Let me know if you have any further trouble, or you want to suggest any changes that make your job easier. Thanks!
